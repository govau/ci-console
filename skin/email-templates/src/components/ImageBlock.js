import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import Table, { Td } from "./Table";
import A from "./A";

const Block = styled(Table)`
  min-width: 100%;
  border-collapse: collapse;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
`;

const BlockInner = styled(Td)`
  padding: 9px;
  mso-line-height-rule: exactly;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
`;

const ContentContainer = styled(Table)`
  min-width: 100%;
  border-collapse: collapse;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
`;

const Content = styled(Td)`
  padding-right: 9px;
  padding-left: 9px;
  padding-top: 0;
  padding-bottom: 0;
  text-align: center;
  mso-line-height-rule: exactly;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
`;

const Img = styled.img`
  padding-bottom: 0;
  display: inline;
  vertical-align: bottom;
  border: 0;
  height: auto;
  outline: none;
  text-decoration: none;
  -ms-interpolation-mode: bicubic;
  width: ${({ width }) => width}px;
  max-width: ${({ maxWidth }) => maxWidth};
`;

const ImageBlock = ({ src, alt, width, maxWidth, link }) => {
  const img = (
    <Img align="center" alt={alt} src={src} width={width} maxWidth={maxWidth} />
  );

  return (
    <Block border="0" cellPadding="0" cellSpacing="0" width="100%">
      <tbody>
        <tr>
          <BlockInner valign="top">
            <ContentContainer
              align="left"
              width="100%"
              border="0"
              cellPadding="0"
              cellSpacing="0"
            >
              <tbody>
                <tr>
                  <Content valign="top">
                    {link ? <A {...link}>{img}</A> : { img }}
                  </Content>
                </tr>
              </tbody>
            </ContentContainer>
          </BlockInner>
        </tr>
      </tbody>
    </Block>
  );
};

ImageBlock.propTypes = {
  src: PropTypes.string.isRequired,
  alt: PropTypes.string.isRequired,
  width: PropTypes.number.isRequired,
  maxWidth: PropTypes.string.isRequired,
  link: PropTypes.shape({
    href: PropTypes.string.isRequired
  })
};

export default ImageBlock;
