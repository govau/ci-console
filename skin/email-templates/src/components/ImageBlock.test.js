import React from "react";
import renderer from "react-test-renderer";

import ImageBlock from "./ImageBlock";

test("renders without crashing", () => {
  const tree = renderer.create(
    <ImageBlock
      src="example.png"
      alt="example.com"
      width={250}
      maxWidth="500px"
      link={{ href: "https://www.example.com" }}
    />
  );
  expect(tree).toMatchSnapshot();
});
