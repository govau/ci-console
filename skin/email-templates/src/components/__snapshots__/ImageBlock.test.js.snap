// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
.c1 {
  mso-table-lspace: 0pt;
  mso-table-rspace: 0pt;
}

.c0 {
  min-width: 100%;
  border-collapse: collapse;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
}

.c2 {
  padding: 9px;
  mso-line-height-rule: exactly;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
}

.c3 {
  min-width: 100%;
  border-collapse: collapse;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
}

.c4 {
  padding-right: 9px;
  padding-left: 9px;
  padding-top: 0;
  padding-bottom: 0;
  text-align: center;
  mso-line-height-rule: exactly;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
}

.c5 {
  padding-bottom: 0;
  display: inline;
  vertical-align: bottom;
  border: 0;
  height: auto;
  outline: none;
  text-decoration: none;
  -ms-interpolation-mode: bicubic;
  width: 250px;
  max-width: 500px;
}

<table
  border="0"
  cellPadding="0"
  cellSpacing="0"
  className="c0 c1"
  width="100%"
>
  <tbody>
    <tr>
      <td
        className="c2"
        valign="top"
      >
        <table
          align="left"
          border="0"
          cellPadding="0"
          cellSpacing="0"
          className="c3 c1"
          width="100%"
        >
          <tbody>
            <tr>
              <td
                className="c4"
                valign="top"
              >
                <a
                  className=""
                  href="https://www.example.com"
                >
                  <img
                    alt="example.com"
                    className="c5"
                    src="example.png"
                    width={250}
                  />
                </a>
              </td>
            </tr>
          </tbody>
        </table>
      </td>
    </tr>
  </tbody>
</table>
`;
